diff --git a/BUILD.gn b/BUILD.gn
index 17bab98e8fa..dbd06ee62dc 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -548,6 +548,8 @@ config("internal_config") {
   if (is_component_build) {
     defines += [ "BUILDING_V8_SHARED" ]
   }
+  #Export v8 symbols by Macro V8_EXPORT.
+  defines += [ "BUILDING_V8_SHARED" ]
 }
 
 # Should be applied to all targets that write trace events.
@@ -4889,6 +4891,8 @@ v8_component("v8_libplatform") {
   if (is_component_build) {
     defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
   }
+  #Export v8 platform symbols by Macro V8_EXPORT.
+  defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
 
   public_configs = [ ":libplatform_config" ]
 
@@ -6411,3 +6415,28 @@ if (!build_with_chromium && v8_use_perfetto) {
     ]
   }
 }  # if (!build_with_chromium && v8_use_perfetto)
+
+v8_component("v8_libwrapper") {
+  sources = [
+    "wrapper/wrapper-inspector.cc",
+    "wrapper/wrapper-console.cc"
+  ]
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+  ]
+  configs = [ ":internal_config" ]
+}
+
+static_library("v8_static") {
+  complete_static_lib = true
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+    ":v8_libwrapper",
+  ]
+}
