diff --git a/src/inspector/v8-debugger-agent-impl.cc b/src/inspector/v8-debugger-agent-impl.cc
index 2e5c8cd417..74fb54be82 100644
--- a/src/inspector/v8-debugger-agent-impl.cc
+++ b/src/inspector/v8-debugger-agent-impl.cc
@@ -518,8 +518,22 @@ Response V8DebuggerAgentImpl::setBreakpointByUrl(
       UNREACHABLE();
   }
   if (breakpoints->get(breakpointId)) {
-    return Response::ServerError(
-        "Breakpoint at specified location already exists.");
+    // return Response::ServerError(
+    //    "Breakpoint at specified location already exists.");
+    // HYBRID ADD BEGIN
+    *outBreakpointId = breakpointId;
+    for (const auto& script : m_scripts) {
+      if (!matches(m_inspector, *script.second, type, selector)) continue;
+      std::unique_ptr<protocol::Debugger::Location> location = setBreakpointImpl(
+          breakpointId, script.first, condition, lineNumber, columnNumber);
+      if (location) {
+        (*locations)->emplace_back(std::move(location));
+      }
+    }
+    // return Response::ServerError(
+    //     "Breakpoint at specified location already exists.");
+    return Response::Success();
+    //HYBRID END
   }
 
   String16 hint;
